### PR TITLE
CP-921: Finish Wiring Notification to Reactor Callback URL

### PR DESCRIPTION
**Pull request by Tankanow**
# Description

This has been manually tested with https://api.cloudzero.com/accounts/v1/link @silvexis developed.

### CODE REVIEW

File: services/connected_account.yaml [hunk 1]
```diff
@@ -172,12 +172,13 @@ Resources:
         - Key: cloudzero-reactor-account-id
           Value: !FindInMap [ CallbackConfiguration, prod, ReactorAccountId ]
       ServiceToken: !GetAtt NotificationFunctionStack.Outputs.Arn
-      AccountId: !Sub ${AWS::AccountId}
+      AccountId: !Ref AWS::AccountId
```
Comment (Tankanow) [inline services/connected_account.yaml]: be consistent here

File: services/connected_account.yaml [hunk 2]
```diff
@@ -172,12 +172,13 @@ Resources:
...
       Version: '1'
       # Parameters from Other Resources in this Stack
       ExternalId: !Ref ExternalId
       ReactorCallbackUrl: !FindInMap [ CallbackConfiguration, prod, ReactorCallbackUrl ]
       AccountName: !Ref AccountName
+      Region: !Ref AWS::Region
```
Comment (Tankanow) [inline services/connected_account.yaml]: `Region` and `LegacyAccount` were missing

File: services/notification/src/app.py [hunk 3]
```diff
@@ -50,6 +51,8 @@
             'ReactorCallbackUrl': str,
             'AccountName': str,
             'ReactorId': str,
+            'AccountId': str,
```
Comment (Tankanow) [inline services/notification/src/app.py]: Add some extra metadata

File: services/notification/src/app.py [hunk 4]
```diff
@@ -157,9 +161,11 @@ def effects_reactor_callback(world):
         **callback_metadata(properties(world)),
         'links': world.get('output', DEFAULT_OUTPUT),
     }
-    response = requests.post(url, data=data)
+    logger.info(f'Posting to {url} this data: {json.dumps(data)}')
```
Comment (Tankanow) [inline services/notification/src/app.py]: some extra logging

File: services/notification/template.yaml [hunk 5]
```diff
@@ -33,10 +33,7 @@ Resources:
         Key: latest/services/notification.zip
       Handler: src.app.handler
       Policies:
-        - SQSSendMessagePolicy:
-            QueueName: !Select
-              - 4
-              - !Split [ '/', !Ref ReactorCallbackUrl ]
+        - AWSCloudFormationReadOnlyAccess
```
Comment (Tankanow) [inline services/notification/template.yaml]: the right policies

File: services/notification/src/app.py [hunk 6]
```diff
@@ -170,9 +176,8 @@ def effects_reactor_callback(world):
 def handler(event, context, **kwargs):
     status = cfnresponse.SUCCESS
     world = {}
-    logger.info('WTF')
     try:
-        logger.info(f'Processing event {event}')
+        logger.info(f'Processing event {json.dumps(event)}')
```
Comment (silvexis) [inline services/notification/src/app.py]: what happens if event isn't JSON, also is there anything sensitive in the event that we might not want to log?

File: services/notification/src/app.py (hunk 4 above)
Comment (silvexis) [inline services/notification/src/app.py]: not sure we want to write the account connection info to the log...

Comment (Tankanow) [reply services/notification/src/app.py]: It's all data about the customer's account. It's definitely helpful for debugging and I thought the customer might like to see the data we're recording. Let me know if you are really think it should be removed.

File: services/notification/src/app.py (hunk 6 above)
Comment (Tankanow) [reply services/notification/src/app.py]: It will be `json.dump`able. Nothing sensitive, it's all data we've discovered about the customer's account.

**Pull request merged**